Bugzilla/TCT.pm

```diff
@@ -0,0 +1,100 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::TCT;
+use 5.10.1;
+use Moo;
+
+use Bugzilla::DaemonControl qw( on_finish on_exception );
+use File::Temp;
+use Future::Utils qw(call);
+use Future;
+use IO::Async::Process;
+
+use constant TCT_BIN => '/usr/local/bin/tct';
+
+has 'public_key' => (
+    is       => 'ro',
+    required => 1,
+);
+
+has 'public_key_file' => (
+    is => 'lazy',
+);
+
+has 'is_valid' => (
+    is => 'lazy',
+);
+
+sub _build_public_key_file {
+    my ($self) = @_;
+    my $fh = File::Temp->new(SUFFIX => '.pubkey');
+    $fh->print($self->public_key);
+    $fh->close;
+    return $fh;
+}
+
+sub _build_is_valid {
+    my ($self) = @_;
+
+    my $loop = IO::Async::Loop->new;
+    my $exit_f = $loop->new_future;
+    my ($stderr, $stdout);
+    my $process = IO::Async::Process->new(
+        command => [TCT_BIN, 'check', '-k', $self->public_key_file ],
```

Comment:
does tct require ascii armoured keys?

the `Crypt::OpenPGP` code explicitly checks for ascii armour, but i don't see a similar check here.
the `textarea` in the UI coupled with the lack of `BINMODE` on the tempfile means a binary key is unlikely to work, but it may be a good idea to add a check for `BEGIN PGP MESSAGE` or similar.

Reply:
non-ascii-armored keys will be rejected. tocotrienol will never support binary format.
The secure mail code wouldn't have ever worked with non-armored keys as well.